src/pathfinding_igvc/include/AStar.h

```diff
@@ -14,15 +15,81 @@
 
 class AStar {
   public:
+    /*
+     * Assumes the value at a cell of occupancy grid is either GRID_FREE
+     * or GRID_OCCUPIED
+     */
     static const int GRID_FREE     = 0;
     static const int GRID_OCCUPIED = 100;
 
+    /*
+     * A representation of a point in grid with its col and row
+     */
     struct GridPoint {
         int col;
         int row;
-        GridPoint(int c = 0, int r = 0) : col(c), row(r){};
+        GridPoint(int r = 0, int c = 0) : col(c), row(r) {};
+    };
+
+    /*
+     * A representation of a point in grid along with its score.
+     * Its score, f, is equal to g+h, where:
+     * g is "the movement cost from the starting point" to this point
+     * h is "the estimated movement cost to move from this point to
+     * the final destination".
+     */
+    typedef std::pair<double, std::pair<int, int>> GridPointWithScore;
+
+    /*
+     * A structure that holds the necessary details for a cell
+     * It stores its parent as well as its score parameters,
+     * f, g, and h.
+     */
+    struct CellDetail
+    {
+        // parent of cell
+        GridPoint parent;
+        // f = g + h
+        double f = FLT_MAX, g = FLT_MAX, h = FLT_MAX;
     };
 
+    /*
+     * parameters for storing the size of the grid,
+     * obtained from the occupancy grid metadata
+     */
+    int _num_rows;
+    int _num_cols;
+
+    /*
+     * parameters for storing the starting and goal
+     * points in the grid
+     */
+    GridPoint _start;
+    GridPoint _goal;
```

Comment:
A lot of these class members seem like they should be private? Is there any reason to have them be public?